client/question/services/SessionHistoryServiceSpec.js

```diff
@@ -63,6 +63,29 @@ describe('SessionHistoryService', function() {
         secondSnapshotIndex)).toEqual('second submission');
     });
 
+    it('should retrieve the correct starter code snapshot', function() {
```

Comment:
Add a test to make sure that it throws an error if you give it an invalid index.